test/test_v2/test_clientapp_funcs.c

```diff
@@ -0,0 +1,404 @@
+                        exit(1);
+                    }
+                    TEST_VERBOSE(("Namespace %s: Rank %d Data validated: Key: %s Value: %s",
+                        myproc->nspace, myproc->rank, key, stringdata));
+                }
+                else if (PMIXT_CHECK_KEY(key, PMIX_JOBID) && val_params->check_pmix_jobid) {
+                    if (0 != strcmp(val_params->pmix_jobid, stringdata)) {
+                        TEST_ERROR(("Key %s failed validation. Get value: %s Validation value: %s",
+                                     key, stringdata, val_params->pmix_jobid));
+                        exit(1);
+                    }
+                    TEST_VERBOSE(("Namespace %s: Rank %d Data validated: Key: %s Value: %s",
+                        myproc->nspace, myproc->rank, key, stringdata));
+                }
+                else if (PMIXT_CHECK_KEY(key, PMIX_LOCAL_PEERS) && val_params->check_pmix_local_peers) {
+                    if (0 != strcmp(val_params->pmix_local_peers, stringdata)) {
+                        TEST_ERROR(("Key %s failed validation. Get value: %s Validation value: %s",
+                                     key, stringdata, val_params->pmix_local_peers));
+                        exit(1);
+                    }
+                    TEST_VERBOSE(("Namespace %s: Rank %d Data validated: Key: %s Value: %s",
+                        myproc->nspace, myproc->rank, key, stringdata));
+                }
+                else if (PMIXT_CHECK_KEY(key, PMIX_HOSTNAME) && val_params->check_pmix_hostname) {
+                    if (0 != strcmp(val_params->pmix_hostname, stringdata)) {
+                        TEST_ERROR(("Key %s failed validation. Get value: %s Validation value: %s",
+                                     key, stringdata, val_params->pmix_hostname));
+                        exit(1);
+                    }
+                    TEST_VERBOSE(("Namespace %s: Rank %d Data validated: Key: %s Value: %s",
+                        myproc->nspace, myproc->rank, key, stringdata));
+                } // other possibilities will be added here later
+                else {
+                    TEST_ERROR(("Check input for case PMIX_STRING: key: %s stringdata: %s",
+                        key, stringdata));
+                    exit(1);
+                }
+                free(stringdata);
+                break;
+            default:
+                TEST_ERROR(("No test logic for type: %d, key: %s", value->type, key));
+                exit(1);
+        }
+    }
+    else {
+        TEST_VERBOSE(("All validation disabled, will not validate: %s", key));
+    }
+    if (!validate_self) {
+        val_params->pmix_local_rank = save_local_rank;
+        val_params->pmix_node_rank = save_node_rank;
+        val_params->pmix_local_size = save_local_size;
+        val_params->pmix_nodeid = save_nodeid;
+        strcpy(val_params->pmix_hostname, save_hostname);
+    }
```

Comment:
Use function-local variables instead of changing the global structure back-and-forth
```suggestion
```

Reply:
Yes. See comment above.